File: apps/mobile/lib/models/material_listing.dart
class MaterialListing {
  final String id;
  final String supplierId;
  final String supplierName;
  final String? supplierPhone;
  final String materialType;
  final String unit;
  final String city;
  final String? district;
  final int price;
  final String description;
  final String status;
  final String createdAt;

  const MaterialListing({
    required this.id,
    required this.supplierId,
    required this.supplierName,
    this.supplierPhone,
    required this.materialType,
    required this.unit,
    required this.city,
    this.district,
    required this.price,
    required this.description,
    required this.status,
    required this.createdAt,
  });

  factory MaterialListing.fromJson(Map<String, dynamic> json) => MaterialListing(
        id: json['id'] as String,
        supplierId: json['supplierId'] as String,
        supplierName: json['supplierName'] as String,
        supplierPhone: json['supplierPhone'] as String?,
        materialType: json['materialType'] as String,
        unit: json['unit'] as String,
        city: json['city'] as String,
        district: json['district'] as String?,
        price: json['price'] as int,
        description: json['description'] as String,
        status: json['status'] as String,
        createdAt: json['createdAt'] as String,
      );
}

File: packages/shared/src/types/material-listing.ts
export type MaterialListingStatus = "AVAILABLE" | "INACTIVE";

export interface MaterialListingDto {
  id: string;
  supplierId: string;
  supplierName: string;
  supplierPhone?: string | null;
  materialType: string;
  unit: string;
  city: string;
  district?: string | null;
  price: number;
  photoUrl?: string | null;
  description: string;
  status: MaterialListingStatus;
  createdAt: string;
}

export interface CreateMaterialListingInput {
  materialType: string;
  city: string;
  district?: string;
  price: number;
  description: string;
}

export interface UpdateMaterialListingInput extends Partial<CreateMaterialListingInput> {
  status?: MaterialListingStatus;
}

export interface MaterialListingSearchQuery {
  materialType?: string;
  city?: string;
  district?: string;
  page?: number;
  pageSize?: number;
}